src/handlers/filter/handler.ts

```diff
@@ -32,7 +32,8 @@ export const handler = async (event: SQSEvent) => {
       try {
         await sendRecord(record);
       } catch (e) {
-        response.batchItemFailures.push({itemIdentifier: record.messageId});
+        const body = JSON.parse(record.body);
+        response.batchItemFailures.push({itemIdentifier: body.messageId});
```

Comment:
I'm revisiting this as part of making the code build and work so it will probably change once we know the format of the message (and how the subscription is set up).

Reply:
I think the object will look like this once we have a raw message subscription
```
{
>   Records: [
>     {
>       body: '{"event_name":"IPV_PASSPORT_CRI_REQUEST_SENT"}',
>       receiptHandle: 'AAA',
>       md5OfBody: 'cfb21792cc00c813b0c6a176643af9d4',
>       eventSourceARN: 'arn:aws:sqs:eu-west-2:000000000000:filter-queue',
>       eventSource: 'aws:sqs',
>       awsRegion: 'eu-west-2',
>       messageId: 'e71c31bf-7871-4806-9daa-597c642a85cc',
>       attributes: [Object],
>       messageAttributes: {}
>     }
>   ]
> }
```
So it will be pulled from the record and not the body. I confused it by adding a messageId to the body as part of the test.